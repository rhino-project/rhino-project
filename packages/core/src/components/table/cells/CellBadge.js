import { Badge } from 'reactstrap';
import { useGlobalComponent } from '../../../hooks/overrides';
import { useTableInheritedProps } from '../../../hooks/table';

export const CellBadgeBase = ({ empty = '-', ...props }) => {
  const { getValue, inheritedProps } = useTableInheritedProps(props);

  if (!getValue()) return <div {...inheritedProps}>{empty}</div>;

  return <Badge {...inheritedProps}>{getValue()}</Badge>;
};

const CellBadge = (props) =>
  useGlobalComponent('CellBadge', CellBadgeBase, props);

export default CellBadge;
