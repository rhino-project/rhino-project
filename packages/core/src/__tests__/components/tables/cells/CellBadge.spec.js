import CellBadge from '../../../../components/table/cells/CellBadge';
import { sharedCellTests } from './sharedCellTests';

describe('CellBadge', () => {
  sharedCellTests(CellBadge);
});
