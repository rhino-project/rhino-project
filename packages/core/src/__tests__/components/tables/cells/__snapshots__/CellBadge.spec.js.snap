// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CellBadge > renders inside of model context 1`] = `
<DocumentFragment>
  <div
    class="model-index-table model-index-table-user"
  >
    <table
      class="table table-striped table-hover"
    >
      <thead>
        <tr>
          <th
            class=""
          >
            <div
              class="d-flex flex-row gap-1"
            >
              <div />
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
        <tr
          class="table-row-clickable"
        >
          <td>
            <div>
              -
            </div>
          </td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <th>
            <div />
          </th>
        </tr>
      </tfoot>
    </table>
  </div>
</DocumentFragment>
`;

exports[`CellBadge > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
